int32 flag
--- 
int32 responseFlag
